backend/populate_catalog/task.py

```diff
@@ -18,11 +30,82 @@
 CATALOG_KEY = "catalog.json"
 CONTENTS_KEY = "Contents"
 RECORDS_KEY = "Records"
+MESSAGE_ATTRIBUTES_KEY = "messageAttributes"
+
+LOGGER = set_up_logging(__name__)
 
 
 def lambda_handler(event: JsonObject, _context: bytes) -> JsonObject:
     """Main Lambda entry point."""
 
+    LOGGER.debug(dumps({EVENT_KEY: event}))
+
+    for message in event[RECORDS_KEY]:
+        if (
+            message[MESSAGE_ATTRIBUTES_KEY][MESSAGE_ATTRIBUTE_TYPE_KEY][STRING_VALUE_KEY_LOWER]
+            == MESSAGE_ATTRIBUTE_TYPE_ROOT
+        ):
+            handle_root(message[BODY_KEY])
+        elif (
+            message[MESSAGE_ATTRIBUTES_KEY][MESSAGE_ATTRIBUTE_TYPE_KEY][STRING_VALUE_KEY_LOWER]
+            == MESSAGE_ATTRIBUTE_TYPE_DATASET
+        ):
+            handle_dataset(message[BODY_KEY])
+        else:
+            raise UnhandledSQSMessageException("Unhandled SQS message type")
+
+    return {}
+
+
+class UnhandledSQSMessageException(Exception):
+    pass
+
+
+class GeostoreSTACLayoutStrategy(HrefLayoutStrategy):
+    def get_catalog_href(self, cat: Catalog, parent_dir: str, is_root: bool) -> str:
+        original_path = cat.get_self_href().split("/")
```

Comment:
As discussed, we should probably use `urlparse(cat.get_self_href()).path` and `rsplit("/", maxsplit=2)` here.